File: RPC1/N.cpp
/*   AUTHOR: julianferres   */
#include <bits/stdc++.h>
using namespace std;

// neal Debugger
template<typename A, typename B> ostream& operator<<(ostream &os, const pair<A, B> &p) { return os << '(' << p.first << ", " << p.second << ')'; }
template<typename T_container, typename T = typename enable_if<!is_same<T_container, string>::value, typename T_container::value_type>::type> ostream& operator<<(ostream &os, const T_container &v) { os << '{'; string sep; for (const T &x : v) os << sep << x, sep = ", "; return os << '}'; }
 
void dbg_out() { cerr << endl; }
template<typename Head, typename... Tail> void dbg_out(Head H, Tail... T) { cerr << ' ' << H; dbg_out(T...); }
 
#ifdef LOCAL
#define dbg(...) cerr << "(" << #__VA_ARGS__ << "):", dbg_out(__VA_ARGS__)
#else
#define dbg(...)
#endif

typedef long long ll;
typedef vector<ll> vi; typedef pair<ll,ll> ii;
typedef vector<ii> vii; typedef vector<bool> vb;
#define FIN ios::sync_with_stdio(0);cin.tie(0);cout.tie(0)
#define forr(i, a, b) for(int i = (a); i < (int) (b); i++)
#define forn(i, n) forr(i, 0, n)
#define DBG(x) cerr << #x << " = " << (x) << endl
#define RAYA cerr << "===============================" << endl
#define pb push_back
#define mp make_pair
#define all(c) (c).begin(),(c).end()
#define esta(x,c) ((c).find(x) != (c).end())
const int INF = 1e9+15; // const ll INF = 2e18;
const int MOD = 1e9+7; // 998244353
const int MAXN  = 2e5+5;

pair<ll,ll> dame(int a){
	int dos=0, cinco=0;
	while(a%2==0){
		a/=2;
		dos++;
	}
	while(a%5==0){
		a/=5;
		cinco++;
	}
	if(a==1) return {dos,cinco};
	else return {-1,-1};
}


int main(){  
    FIN;

    int a, n;
    while(cin >> a){
    	cin >> n;
    	ii ans = dame(a);
    	if(ans == -1){
    		cout << "Precision Error\n";
    	} else{
    		if(ans.first > ans.second){
    			
    		} 
    		else if(ans.first < ans.second){

    		} 
    		else{
    			cout << 1 << "\n";
    		}
    	}

    }
    

    return 0;
}

File: RPC1/N.py

import sys

def dame(a):
	dos, cinco = 0, 0
	while(a % 2 == 0):
		a //= 2
		dos += 1
	while(a % 5 == 0):
		a //= 5
		cinco += 1

	if(a == 1):
		return (dos, cinco)
	else:
		return (-1, -1)


def be(a, b):
    res = 1
    while (b > 0):
        if (b & 1):
            res = res * a
        a = a * a
        b >>= 1
    return res;


def main():
	#Leer de archivo
	for line in sys.stdin:
   		a, n = map(int, line.split())

   		ans = dame(a)

   		if(ans[0] == -1 and n > 0):
   		 	print("Precision Error")
   		else:
   			print(max(ans[0]*n, ans[1]*n), be(2,(max(ans)*n - ans[0]*n)) * be(5, (max(ans)*n - ans[1]*n)))

main()

